src/mon/NVMeofGwSerialize.h

```diff
@@ -407,28 +480,43 @@ inline void decode(
   NvmeGwMonStates& gws, ceph::buffer::list::const_iterator &bl) {
   gws.clear();
   uint32_t num_created_gws;
-  DECODE_START(1, bl);
+  DECODE_START(2, bl);
+  dout(20) << "decode NvmeGwMonStates. struct_v: " << struct_v << dendl;
   decode(num_created_gws, bl);
-
+  dout(20) << "decode NvmeGwMonStates. num gws  " << num_created_gws << dendl;
+  uint8_t anas[MAX_SUPPORTED_ANA_GROUPS];         // for old decode
+  for (uint32_t i = 0; i <MAX_SUPPORTED_ANA_GROUPS; i ++) anas[i] = 0;
   for (uint32_t i = 0; i<num_created_gws; i++) {
     std::string gw_name;
     decode(gw_name, bl);
     NvmeAnaGrpId ana_grp_id;
     decode(ana_grp_id, bl);
-
+    dout(20) << "decode NvmeGwMonStates. GW-id " << gw_name << " ana grpid "<< ana_grp_id <<  dendl;
+    anas[ana_grp_id] = 1;
```

Comment:
Also, what happens if the id happens to be REDUNDANT (0xFF)?  Or is that disallowed by some other means?

Reply:
REDUNDANT  is disallowed probably would be removed in some commit

Reply:
I think for SQUID in order to be backward compatible with REEF we are not supporting more than 8 GWs in the same 
<pool group>  But  if someone configured wrong number gws  it can be a memory corruption,  agree

Reply:
AFAIK in Reef we soft limited to 4 GWs, but the code is able to support up to 16 (which I think is the MAX_SUPPORTED_ANA_GROUPS?). 
In Squid I do want to support up to 16.

Reply:
Where is that limit enforced in the code?

Reply:
I think we might need to add it. Like when doing "create-gw" we should fail if number of gw in this {pool, group} is already 16. @leonidc do you agree?

Reply:
I agree , otherwise we cannot guarantee HA during upgrade